### PR TITLE
Use Gulp and Browserify

### PR DESCRIPTION
**Pull request by trotod**
Here is what I found:

+ When developing, Gulp is _extremely_ fast.
+ ~~Compile for production speeds, however, are slower than Grunt. (Takes about 1 min, so I suggest just compiling before copying files to a static file server). About 80-90% of that time is due to revving and minimizing assets.~~ EDIT: This is fixed. Now much faster than Grunt (~ 7 seconds) when building.

Basically just `npm install -g gulp` and `npm install` to get the new dependencies.

Browserify, on the other hand, makes JavaScript easier to write, but you guys might not like it.

### CODE REVIEW

**Comment by trotod** [comment]
I think the time can be minimized if I can find a way to get the `html:dist` task to only have to read one html file instead of (I don't know how many...).

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+  var jsFilter = $.filter('**/*.js')
+    , cssFilter = $.filter('**/*.css')
+    , htmlFilter = $.filter('**/*.html')
+    , indexFilter = $.filter('index.html')
+
+  var assets = $.useref.assets({ searchPath: '{.tmp,app}' })
+
+  return gulp.src('dist/**/*.html')
+    .pipe(indexFilter) // only source from index.html
```
Comment (trotod) [inline gulpfile.js]: Production speeds now increased drastically due to sourcing of only index.html when looking for assets.

**Comment by trotod** [comment]
Should I merge this myself?

**Comment by shovikguha** [comment]
Not until we have the site up.

On Sun, Jan 4, 2015 at 1:54 PM, David Liu <notifications@github.com> wrote:

> Should I merge this myself?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/shovikguha/MVRT_Site/pull/12#issuecomment-68650984>.
>



-- 
/*no comment*/

**Comment by trotod** [comment]
I don't see why not...nothing really changes besides the command name (`gulp`) and JavaScripts no one really edits.

Plus we get super fast build times now.

**Comment by m3rcuriel** [comment]
I'll start testing

**Pull request merged**